tests/allowed-signers.rs

```diff
@@ -0,0 +1,44 @@
+use sshcerts::ssh::AllowedSigners;
+
+#[test]
+fn parse_good_allowed_signers() {
+    let allowed_signers = AllowedSigners::from_path("tests/allowed_signers/good_allowed_signers");
+    assert!(allowed_signers.is_ok());
+    let AllowedSigners(allowed_signers) = allowed_signers.unwrap();
+    assert_eq!(allowed_signers.len(), 3);
+
+    assert_eq!(
+        allowed_signers[0].key.fingerprint().to_string(),
+        "SHA256:QAtqtvvCePelMMUNPP7madH2zNa1ATxX1nt9L/0C5+M",
+    );
+    assert_eq!(
+        allowed_signers[0].principals,
+        vec!["mitchell@confurious.io".to_string()],
+    );
+    assert!(!allowed_signers[0].cert_authority);
+    assert!(allowed_signers[0].namespaces.is_none());
+    assert!(allowed_signers[0].valid_after.is_none());
+    assert!(allowed_signers[0].valid_before.is_none());
+
+    assert_eq!(
+        allowed_signers[1].principals,
+        vec!["mitchell@confurious.io".to_string(), "mitchel2@confurious.io".to_string()],
+    );
+    assert!(allowed_signers[1].cert_authority);
+    assert_eq!(
+        allowed_signers[1].namespaces, 
+        Some(vec!["thanh".to_string(), "#mitchell".to_string()])
+    );
+    assert!(allowed_signers[1].valid_after.is_none());
+    assert_eq!(allowed_signers[1].valid_before, Some(123u64));
+
+    assert_eq!(
+        allowed_signers[2].namespaces, 
+        Some(vec![
+            "thanh".to_string(),
+            " ".to_string(),
+            "mitchell mitchell".to_string(),
+            " andrew   andrew".to_string(),
+        ]),
+    );
```

Comment:
Apparently these are allowed. I was testing it against the git implementation. It's quite confusing.

> [namespaces](https://man.openbsd.org/ssh-keygen.1#namespaces)=namespace-list
Specifies a pattern-list of namespaces that are accepted for this key. If this option is present, the signature namespace embedded in the signature object and presented on the verification command-line must match the specified list before the key will be considered acceptable.
https://man.openbsd.org/ssh-keygen.1#ALLOWED_SIGNERS

So namespaces-list is a pattern-list. 

But the [definition of pattern-list](https://man.openbsd.org/ssh_config.5#PATTERNS) doesn't really allow whitespaces. But the parser doesn't reject whitespaces: https://github.com/openssh/openssh-portable/blob/385ecb31e147dfea59c1c488a1d2011d3867e60e/sshsig.c#L889, same with whatever implementation git uses.

Reply:
The openssh implementation parses `namespaces` into a String instead of Vec<String>. And it uses this String to do direct pattern matching against concrete namespaces.